code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCardEventObserver.kt

```diff
@@ -0,0 +1,184 @@
+        }
+
+        when (event) {
+            is UIEvent.Display -> {
+                if (callback?.onCardDisplayed(event) == true) {
+                    if (event.aepUi.getTemplate() is SmallImageTemplate) {
+                        Log.trace(MessagingConstants.LOG_TAG, SELF_TAG, "SmallImageUI Displayed")
+                        handleSmallImageTracking(
+                            event.aepUi.getTemplate() as SmallImageTemplate,
+                            null,
+                            MessagingEdgeEventType.DISPLAY
+                        )
+                    }
+                }
+            }
+            is UIEvent.Interact -> {
+                handleInteractEvent(event)
+            }
+            is UIEvent.Dismiss -> {
+                if (callback?.onCardDismiss(event) == true) {
+                    if (event.aepUi.getTemplate() is SmallImageTemplate) {
+                        Log.trace(MessagingConstants.LOG_TAG, SELF_TAG, "SmallImageUI Dismissed")
+                        handleSmallImageTracking(event.aepUi.getTemplate() as SmallImageTemplate, null, MessagingEdgeEventType.DISMISS)
+                    }
+                }
+            }
+        }
+    }
+
+    private fun handleInteractEvent(event: UIEvent.Interact<*, *>) {
+        when (event.action) {
+            UIAction.CLICK -> {
+                if (event.aepUi.getTemplate() is SmallImageTemplate) {
+                    handleSmallImageClickInteraction(event)
+                }
+            }
+            UIAction.DOUBLE_CLICK -> {
+                // TODO: handle double click action
+            }
+            UIAction.DRAG -> {
+                // TODO: handle drag action
+            }
+            UIAction.SWIPE -> {
+                // TODO: handle swipe action
+            }
+            UIAction.LONG_PRESS -> {
+                // TODO: handle long press action
+            }
+            UIAction.EXPAND -> {
+                // TODO: handle expand action
+            }
+        }
+    }
+
+    private fun handleSmallImageClickInteraction(event: UIEvent.Interact<*, *>) {
+        val smallImageTemplate = (event.aepUi.getTemplate() as SmallImageTemplate)
+        // content card buttons have an interact id, dismiss buttons will have the style as the interact id
+        if (!event.interactId.isNullOrEmpty()) {
+            val contentCardSchemaData = ContentCardMapper.instance.getContentCardSchemaDataForPropositionId(smallImageTemplate.id)
+            when (event.interactId) {
+                "none", "simple", "circle" -> { // handle content card dismiss button click
+                    if (callback?.onDismissButtonClick(event) == true) {
+                        Log.trace(MessagingConstants.LOG_TAG, SELF_TAG, "SmallImageUI dismiss button click")
+                        contentCardSchemaData?.track(event.action.toString(), MessagingEdgeEventType.INTERACT)
+                    }
+                }
```

Comment:
Wouldn't interaction with dismiss button produce `UIEvent.Dismiss` instead of `UIEvent.Click`?

Reply:
you are right, fixed this